### DISCUSSION TITLE
Custom driver

### DISCUSSION THREAD

**Discussion by matze1610** [mikro-orm/mikro-orm#1783] [Q&A] [answered]
Hi Martin,

Request/Question:
We plan to provide a custom driver for our database, a HANA in the current case.

One question already popped up: How can we add the new driver to the list of drivers so that we can specify it in the options upon MikroORM.init (other than modifying the source)? 
Right now the list of drivers is a fixed list...

Thanks
  Mathias

PS. Do you have any tricks / tips in your sleeve for implementing custom drivers?

**Comment by B4nan** [comment, marked as answer]
You can pass the driver class to the options:

```ts
import { MyCustomDriver } from './MyCustomDriver.ts';

const orm = await MikroORM.init({
  entities: [Author, Book, ...],
  dbName: 'my-db-name',
  driver: MyCustomDriver, // provide the class, not just its name
});
```

https://mikro-orm.io/docs/usage-with-sql/#custom-driver

Only advice I can give is to look at the existing drivers. If its SQL based, then the best way is to build on top of the knex driver (`AbstractSqlDriver`), if not, look at how mongo driver works.

**Reply by matze1610** [reply]
Sure, thanks:) We'll do!
Have a nice day!
